File: nodes/has-object-property/has-object-property.process.js
/**
 * Process function for the Has Object Property node.
 * Check if an object has a specific property/key.
 */
export default async ({inputs, settings, config}) => {
  const { object, key } = inputs;
  const { check_own_properties } = settings;
  
  // Validate inputs
  if (!object || typeof object !== 'object') {
    return {
      has_property: false,
      key: key || '',
      value: null
    };
  }
  
  if (!key || typeof key !== 'string') {
    return {
      has_property: false,
      key: key || '',
      value: null
    };
  }
  
  // Check if property exists
  let has_property;
  if (check_own_properties) {
    has_property = Object.prototype.hasOwnProperty.call(object, key);
  } else {
    has_property = key in object;
  }
  
  const value = has_property ? object[key] : null;
  
  return {
    has_property: has_property,
    key: key,
    value: value
  };
};

File: nodes/has-object-property/has-object-property.process.py
"""
Process function for the Has Object Property node.
Check if an object has a specific property/key.
"""
from typing import Any, Dict

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    object_input = inputs.get("object")
    key = inputs.get("key")
    check_own_properties = settings.get("check_own_properties", True)
    
    # Validate inputs
    if not isinstance(object_input, dict):
        return {
            "has_property": False,
            "key": key or "",
            "value": None
        }
    
    if not isinstance(key, str):
        return {
            "has_property": False,
            "key": key or "",
            "value": None
        }
    
    # Check if property exists
    has_property = key in object_input
    value = object_input.get(key) if has_property else None
    
    return {
        "has_property": has_property,
        "key": key,
        "value": value
    }

